### PR TITLE
Fix/setup command

### LINKED ISSUE #209: Variants setup on M1 mac does not automatically link variants.xcconfig to xcodeproject
### Environment  ###
**Variants:**
 - Way of installation: make install, v3.6.18
 - Version: 1.1.3

**Your machine:**
 - OS:  MacOS
 - Processor: Apple M1 Pro

**Project's platform:**
 - Platform: iOS

### Describe the bug ###
After successfully running `variants setup` I expect to see `variants.xcconfig` selected at the configuration for my Xcode project under the info tab, for both `debug` and `release`

**Steps to reproduce**
1. Create new iOS project
2. Create variants.yml in root folder
3. install Variants
4. Execute command `variants setup`
5. See error

**Expected behavior**
`variants.xcconfig` successfully linked to my Xcode project under Configurations and I should be able to see the variables defined in `variants.xcconfig` under under Build Settings>User Defined in Xcode

### PR DESCRIPTION
**Pull request by arthurpalves**
### What does this PR do
- Ensures a Xcode project is properly modified when running the command `variants setup`.

### How can it be tested
Among other things:
- Run `variants setup` on a new or existing project that doesn't use Variants yet.
- The project should now contain a `Variants/` folder.
- The project should now use `variants.xcconfig` for its `debug` and `release` configurations.
- The file `project.pbxproj` should have changes pointing default values to values specified in `variants.xcconfig`.
- The project's icon should match the one specified in `variants.yml`.

### Task
resolves #209 

### Checklist:

- [x] I ran `make validation` locally with success
- [x] I have not introduced new bugs
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new errors

### CODE REVIEW

**Comment by backbaseoss** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Pull request is relatively big. If this PR contains multiple changes,
consider splitting it into separate PRs for easier reviews.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 4947a1a99a0d053f87a7429c72cb0c023044992a
</p>

**Comment by arthurpalves** [comment]
> Warnings
> ⚠️	Pull request is relatively big. If this PR contains multiple changes, consider splitting it into separate PRs for easier reviews.
> Generated by 🚫 [Danger Swift](https://danger.systems/swift) against [4947a1a](https://github.com/Backbase/variants/commit/4947a1a99a0d053f87a7429c72cb0c023044992a)

These changes are related to the sample/test app, nothing to be done here

File: Sources/VariantsCore/Factory/iOS/XCConfigFactory.swift [hunk 1]
```diff
@@ -127,7 +127,7 @@ class XCConfigFactory: XCFactory {
                             target: NamedTarget,
                             variant: iOSVariant) {
         let variantsFile = Path("\(xcConfigFile.parent().absolute().description)/Variants.swift")
-
+
```
Comment (GMinucci) [inline Sources/VariantsCore/Factory/iOS/XCConfigFactory.swift:130]: Is our linter enabled for trailing whitespaces?

File: Sources/VariantsCore/Factory/iOS/XcodeProjFactory.swift [hunk 2]
```diff
@@ -190,32 +195,46 @@ struct XcodeProjFactory {
...
-    private func varientsGroup(for projectPath: Path, sourceRoot: Path, target: NamedTarget) throws -> PBXGroup?{
-        let variantsGroupPath = Path("\(projectPath)/Variants")
-        let project = try XcodeProj(path: projectPath)
-    
+    private func createVarientsGroup(
+        for project: XcodeProj,
+        path: Path,
+        sourceRoot: Path,
+        target: NamedTarget
+    ) throws -> PBXGroup?{
```
Comment (GMinucci) [inline Sources/VariantsCore/Factory/iOS/XcodeProjFactory.swift:203]:
```suggestion
    ) throws -> PBXGroup? {
```

File: Sources/VariantsCore/Factory/iOS/XcodeProjFactory.swift [hunk 3]
```diff
@@ -224,7 +243,7 @@ private extension XcodeProjFactory {
          * If .xcconfig, set baseConfigurationReference to it
          */
         if file.lastComponent.contains(".xcconfig"), let fileReference = fileRef {
```
Comment (GMinucci) [inline Sources/VariantsCore/Factory/iOS/XcodeProjFactory.swift:245]: Not really a change you did, but it seems we can:
```suggestion
        if file.extension == "xcconfig", let fileReference = fileRef {
```
https://github.com/kylef/PathKit/blob/master/Sources/PathKit.swift#L256

File: Sources/VariantsCore/Factory/iOS/XCConfigFactory.swift (hunk 1 above)
Comment (arthurpalves) [reply Sources/VariantsCore/Factory/iOS/XCConfigFactory.swift:130]: Double checked and `trailing_whitespace` is disabled. I can tackle this in a swiftlint specific PR

**Review by GMinucci — APPROVED** [review]

**Pull request merged**